### PR TITLE
REST endpoints for osdconfig

### PR DESCRIPTION
**Pull request by sdeoras**
**What this PR does / why we need it**:
Added REST endpoints for osdconfig

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/client/cluster/client.go [hunk 1]
```diff
@@ -277,3 +280,54 @@ func (c *clusterClient) EraseAlert(resource api.ResourceType, alertID int64) err
...
+	}
+	return config, nil
+}
+
+func (c *clusterClient) SetClusterConf(config *osdconfig.ClusterConfig) error {
+	data, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+	request := c.c.Post().Body(data).Resource(clusterPath + "/config/cluster")
```
Comment (lpabon) [inline api/client/cluster/client.go]: Try to make these constants

File: api/server/cluster.go [hunk 2]
```diff
@@ -41,7 +44,159 @@ func (c *clusterApi) Routes() []*Route {
 		{verb: "GET", path: clusterPath("/alerts/{resource}", cluster.APIVersion), fn: c.enumerateAlerts},
 		{verb: "PUT", path: clusterPath("/alerts/{resource}/{id}", cluster.APIVersion), fn: c.clearAlert},
 		{verb: "DELETE", path: clusterPath("/alerts/{resource}/{id}", cluster.APIVersion), fn: c.eraseAlert},
+		{verb: "GET", path: clusterPath("/config/cluster", cluster.APIVersion), fn: c.getClusterConf},
+		{verb: "GET", path: clusterPath("/config/node/{id}", cluster.APIVersion), fn: c.getNodeConf},
```
Comment (lpabon) [inline api/server/cluster.go]: The constants will match these strings

File: cluster/manager.go [hunk 3]
```diff
@@ -1799,3 +1806,28 @@ func (c *ClusterManager) putNodeCacheEntry(nodeId string, node api.Node) {
...
+
+func (c *ClusterManager) SetClusterConf(config *osdconfig.ClusterConfig) error {
+	return c.configManager.SetClusterConf(config)
+}
+
+func (c *ClusterManager) SetNodeConf(config *osdconfig.NodeConfig) error {
+	return c.configManager.SetNodeConf(config)
+}
+
+func (c *ClusterManager) WatchCluster(name string, cb func(config *osdconfig.ClusterConfig) error) error {
```
Comment (lpabon) [inline cluster/manager.go]: Create appropriate interfaces so that this object does not need to satisfy functions which will not be used.

Comment (sdeoras) [reply cluster/manager.go]: done

**Pull request merged**